src/ai/backend/manager/repositories/container_registry/admin_repository.py

```diff
@@ -0,0 +1,49 @@
+from typing import Optional
+
+from ai.backend.manager.data.container_registry.types import ContainerRegistryData
+from ai.backend.manager.models.utils import ExtendedAsyncSAEngine
+
+from .repository import ContainerRegistryRepository
+
+
+class AdminContainerRegistryRepository:
```

Comment:
It would be good to add a permission check hook that runs before the method is executed.